src/switch_jitterbuffer.c

```diff
@@ -854,12 +913,128 @@ static inline switch_status_t jb_next_packet_by_ts(switch_jb_t *jb, switch_jb_no
 
 }
 
+static inline int check_jb_size(switch_jb_t *jb)
+{
+	switch_jb_node_t *np;
+	uint16_t seq;
+	uint16_t l_seq=0;
+	uint16_t h_seq=0;
+	uint16_t count=0;
+	uint16_t old=0;
+	switch_mutex_lock(jb->list_mutex);
+
+	for (np = jb->node_list; np; np = np->next) {
+		if (!np->visible) {
+			continue;
+		}
+
+	seq = ntohs(np->packet.header.seq);
+	if (ntohs(jb->target_seq) > seq) {
+		hide_node(np, SWITCH_FALSE);
+		old++;
+		continue;
+	}
+	if (count == 0) {
+		l_seq = h_seq = seq;
+	}
+	count++;
+	if (seq < l_seq)
+		l_seq = seq;
+	if (seq > h_seq)
+		h_seq = seq;
+	}
+	if (count > jb->jitter.stats.size_max) {
+		jb->jitter.stats.size_max = count;
+	}
+	if (jb->jitter.stats.size_est == 0) {
+		jb->jitter.stats.size_est = count;
+	} else {
+		jb->jitter.stats.size_est = ((99*jb->jitter.stats.size_est)+(1*count))/100;
+	}
+	if (ntohs(jb->target_seq) % 50 == 0) { /* update the stats every x packets */
```

Comment:
Modulo operations are really slow and should be avoided if possible. if `% 50` can be `% 64` or `% 32` then we could convert that into an `AND` of a power of 2. Here is an example: `a % 64` equals `a & 63` (`AND` is a very fast operation)

Reply:
20 ms ptime * 50 packets = 1 second of audio. for clarity, it's better to get these stats every 1 second, not every 1.28 seconds.